subsys/bluetooth/common/Kconfig

```diff
@@ -184,11 +184,34 @@ config BT_BUF_CMD_TX_SIZE
 
 config BT_BUF_CMD_TX_COUNT
 	int "Number of HCI command buffers"
-	default 2
-	range 2 64
+	default 2 if BT_HCI_ACL_FLOW_CONTROL
+	default 1
+	range 1 64
 	help
 	  Number of buffers available for outgoing HCI commands from the Host.
 
+	  HCI Controllers may not support Num_HCI_Command_Packets > 1, hence
+	  default to 1 when not enabling Controller to Host data flow control,
+	  BT_HCI_ACL_FLOW_CONTROL.
+
+	  Normal HCI commands follow the HCI command flow control using
+	  Num_HCI_Command_Packets return in HCI command complete and status.
+
+	  Host Number of Completed Packets command does not follow normal flow
+	  control of HCI commands and the Controller side HCI drivers that
+	  allocates HCI command buffers with K_NO_WAIT can end up running out
+	  of command buffers.
+
+	  When Controller to Host data flow control is enabled in the Host only
+	  build, ensure that BT_BUF_CMD_TX_COUNT is at least 2, to be able to
+	  send one normal HCI command and one Host Number of Completed Packets
+	  command.
+
+	  When Controller to Host data flow control is supported in the
+	  Controller only build, ensure that BT_BUF_CMD_TX_COUNT is greater than
+	  or equal to (BT_BUF_ACL_RX_COUNT + Ncmd), where Ncmd is supported
+	  maximum Num_HCI_Command_Packets in the Controller implementation.
```

Comment:
I don't think this can be enforced via Kconfig, but we can have BUILD_ASSERTs for it

Reply:
Yes, added runtime check on Host Buffer Size command and will derive for Controller-only (not allow it to be configured incorrectly).